plugin/storage/es/factory.go

```diff
@@ -74,16 +78,58 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 // CreateSpanReader implements storage.Factory
 func (f *Factory) CreateSpanReader() (spanstore.Reader, error) {
 	cfg := f.primaryConfig
-	return esSpanStore.NewSpanReader(f.primaryClient, f.logger, cfg.GetMaxSpanAge(), f.metricsFactory, cfg.GetIndexPrefix()), nil
+	return esSpanStore.NewSpanReader(esSpanStore.SpanReaderParams{
+		Client:            f.primaryClient,
+		Logger:            f.logger,
+		MetricsFactory:    f.metricsFactory,
+		MaxLookback:       cfg.GetMaxSpanAge(),
```

Comment:
@pavolloffay why did we rename it? It's confusing that config params do not match the storage params.

Reply:
I don't think we renamed anything in this PR

The flag name is `--es.max-span-age` and configuration `cfg.GetMaxSpanAge()`. `MaxLookback` is only used in reader maybe also writer.

Reply:
I think we should change everything to maxSpanAge

Reply:
I can change it, there are also other refactoring opportunities